test/corpus/todo.txt

```diff
@@ -0,0 +1,486 @@
+============
+comment char
+============
+
+'#'
+
+---
+
+(source_file
+  (char
+    (comment)
+    (MISSING "'")))
+
+================
+include + extend
+================
+
+class MyClass
+  include MyModule
+  extend MyModule
+end
+
+---
+
+(source_file
+  (class_def
+    (constant)
+    (call
+      (identifier)
+      (argument_list
+        (constant)))
+    (call
+      (identifier)
+      (argument_list
+        (constant)))))
+
+================
+select statement
+================
+
+select thing
+when channel.receive
+  puts "hello"
+else
+  puts "world"
+end
+
+---
+
+(source_file
+  (ERROR
+    (call
+      (identifier)
+      (argument_list
+        (identifier)))
+    (call
+      (identifier)
+      (identifier))
+    (call
+      (identifier)
+      (argument_list
+        (string))))
+  (call
+    (identifier)
+    (argument_list
+      (string)))
+  (ERROR))
+
+=========
+macro def
+=========
+
+macro name
+end
+
+---
+
+(source_file
+  (call
+    (identifier)
+    (argument_list
+      (identifier)))
+  (ERROR))
+
+======
+macros
+======
+
+{% if true %}
+{% for i in [1, 2, 3] %}
+%var = 1 + {{ word }}
+foo = %var + 4
+{% end %}
+{% end %}
+
+---
+
+(ERROR
+  (MISSING identifier)
+  (operator)
+  (op_call
+    (op_call
+      (true)
+      (ERROR
+        (operator))
+      (operator)
+      (call
+        (identifier)
+        (argument_list
+          (call
+            (identifier)
+            (argument_list
+              (call
+                (identifier)
+                (argument_list
+                  (array
+                    (integer)
+                    (integer)
+                    (integer)))))))))
+    (ERROR
+      (operator))
+    (operator)
+    (assign
+      (identifier)
+      (op_call
+        (integer)
+        (operator)
+        (tuple
+          (tuple
+            (identifier))))))
+  (then
+    (op_call
+      (op_call
+        (identifier)
+        (ERROR)
+        (operator)
+        (identifier))
+      (operator)
+      (integer))
+    (tuple
+      (op_call
+        (op_call
+          (MISSING identifier)
+          (operator)
+          (identifier))
+        (operator)
+        (MISSING identifier)))
+    (tuple
+      (op_call
+        (op_call
+          (MISSING identifier)
+          (operator)
+          (identifier))
+        (operator)
+        (MISSING identifier)))))
+
+=================
+lib def + objects
+=================
+
+lib MyLib
+  fun name = Name : Void
+
+  fun cos(value : Float64) : Float64
+
+  union IntOrFloat
+    some_int : Int32
+    some_float : Float64
+  end
+
+  struct TimeZone
+    minutes_west : Int32
+    dst_time : Int32
```

Comment:
Not a requirement but might be worth including external vars too, e.g. `$ext : Void*`.

Reply:
Wow had no idea these existed, good point
https://crystal-lang.org/reference/1.14/syntax_and_semantics/c_bindings/variables.html